test/test_config.c

```diff
@@ -0,0 +1,99 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include "axl.h"
+#include "axl_internal.h"
+
+#include "kvtree.h"
+#include "kvtree_util.h"
+
+int
+main(void) {
+    int rc;
+    char *state_file = NULL;
+    kvtree* axl_config_values = kvtree_new();
+
+    size_t old_axl_file_buf_size = axl_file_buf_size;
+    int old_axl_debug = axl_debug;
+    int old_axl_make_directories = axl_make_directories;
+    int old_axl_copy_metadata = axl_copy_metadata;
+
+    rc = AXL_Init(state_file);
+    if (rc != AXL_SUCCESS) {
+        printf("AXL_Init() failed (error %d)\n", rc);
+        return rc;
+    }
+
+    /* check AXL configuration settings */
+    rc = kvtree_util_set_bytecount(axl_config_values,
+                                   AXL_KEY_CONFIG_FILE_BUF_SIZE,
+                                   old_axl_file_buf_size + 1);
+    if (rc != KVTREE_SUCCESS) {
+        printf("kvtree_util_set_bytecount failed (error %d)\n", rc);
+        return rc;
+    }
+    rc = kvtree_util_set_int(axl_config_values, AXL_KEY_CONFIG_DEBUG,
+                             !old_axl_debug);
+    if (rc != KVTREE_SUCCESS) {
+        printf("kvtree_util_set_int failed (error %d)\n", rc);
+        return rc;
+    }
+    rc = kvtree_util_set_int(axl_config_values, AXL_KEY_CONFIG_MKDIR,
+                             !old_axl_make_directories);
+    if (rc != KVTREE_SUCCESS) {
+        printf("kvtree_util_set_int failed (error %d)\n", rc);
+        return rc;
+    }
+    rc = kvtree_util_set_int(axl_config_values, AXL_KEY_CONFIG_COPY_METADATA,
+                             !old_axl_copy_metadata);
+    if (rc != KVTREE_SUCCESS) {
+        printf("kvtree_util_set_int failed (error %d)\n", rc);
+        return rc;
+    }
+
+    printf("Configuring AXL...\n");
+    if (AXL_Config(axl_config_values) == NULL) {
+        printf("AXL_Config() failed\n");
+        return EXIT_FAILURE;
+    }
```

Comment:
You need to AXL_Create() a transfer and add tests for per-ID config options.

Reply:
At this point there are no per-ID config options so nothing to test.

Reply:
`axl_file_buf_size` should be per-ID